agents/agents/guard/fraud.go

```diff
@@ -101,7 +123,17 @@ func (g Guard) handleSnapshot(ctx context.Context, log ethTypes.Log) error {
 // Only submit a state report if we are not on Summit, and the snapshot
 // agent is not currently in dispute.
 func (g Guard) shouldSubmitStateReport(ctx context.Context, snapshot *types.FraudSnapshot) (bool, error) {
-	disputeStatus, err := g.domains[g.summitDomainID].BondingManager().GetDisputeStatus(ctx, snapshot.Agent)
+	var disputeStatus types.DisputeStatus
+	var err error
+	contractCall := func(ctx context.Context) error {
```

Comment:
At some point we want to allow configurable backoff logic. In the `Guard` we could do something like:

```go
g.backoffStrat = g.cfg.MaxRetryDurationSeconds
```

And we could make the contract call like this:

```go
err = retry.WithBackoff(ctx, contractCall)
```

Up to you if we do it in this PR or merge this one and impl the backoff config in the base PR!

Reply:
just added